src/main/scala-2.12/org/ergoplatform/appkit/ergotool/dex/CancelOrderCmd.scala

```diff
@@ -109,12 +109,20 @@ object CancelOrder {
   }
 
   // TODO: extract to use in other commands
-  case class OutBoxProto(getValue: Long, tokens: Seq[ErgoToken], contract: ErgoContract) {
+  case class OutBoxProto(getValue: Long, tokens: Seq[ErgoToken], registers: Seq[ErgoValue[_]], contract: ErgoContract) {
```

Comment:
I don't see enough motivation to introduce this class. Or please describe the role of the class in ScalaDoc.
Same for TxProto, especially if it is going to be reusable.

Reply:
Also, let's don't pollute the code with Try everywhere. 
This code really sucks both from readability point of view and form error handling and reporting.
```
val tx = CancelOrder.createTx(orderBox, recipientAddress, unspentBoxesForAmount)
        .fold(t => error(s"$t"), tx => tx).toTx(ctx.newTxBuilder)
```
Please refactor createTx to return normal object, it may throw error and it will be catched up the stack as necessary.

Reply:
> Please refactor createTx to return normal object, it may throw error and it will be catched up the stack as necessary.

Done.

Reply:
> I don't see enough motivation to introduce this class. Or please describe the role of the class in ScalaDoc.
> Same for TxProto, especially if it is going to be reusable.

These classes were a raw quick hack so that I could extract the algo for this command from `ergoClient.execute` into "effect-less" `CancelOrder` and test the algo without tx and box builders in [tests](http://github.com/aslesarenko/ergo-tool/blob/2e331299169613e5d3597b9ce497b78e301b874f/src/test/scala/org/ergoplatform/appkit/ergotool/dex/CancelOrderSpec.scala#L23-L95). 

With the current approach to command design, I'm not sure they are much sense to re-use them.
What do you think of command design with effects parametrized and using case classes for domain models(tx, boxes, etc.)?